src/Nethermind/Nethermind.Merge.Plugin/PoSSwitcher.cs

```diff
@@ -151,29 +151,6 @@ public bool TryUpdateTerminalBlock(BlockHeader header)
             return true;
         }
 
-        public void ForkchoiceUpdated(BlockHeader newHeadHash, Keccak finalizedHash)
-        {
-            if (finalizedHash != Keccak.Zero && _finalizedBlockHash == Keccak.Zero)
-            {
-                _blockTree.NewHeadBlock -= CheckIfTerminalBlockReached;
-            }
-
-            if (finalizedHash != Keccak.Zero)
-            {
-                if (_finalizedBlockHash == Keccak.Zero)
-                {
-                    if (_logger.IsInfo)
-                        _logger.Info(
-                            $"Reached the first finalized PoS block FinalizedHash: {finalizedHash}, NewHeadHash: {newHeadHash}");
-                    _blockTree.NewHeadBlock -= CheckIfTerminalBlockReached;
-                }
-
-                _finalizedBlockHash = finalizedHash;
-                // ToDo need to discuss with Sarah, this should be moved to BlockTree or FinalizationManager
```

Comment:
Remove metadata save here not whole method. Plus remove one line from ForkchoiceUpdatedHanlder as I mentioned in previous PR